core-plugins/src/main/java/io/cdap/plugin/batch/source/FTPBatchSource.java

```diff
@@ -141,7 +141,7 @@ public FileFormat getFormat() {
     @Nullable
     @Override
     public Pattern getFilePattern() {
-      return null;
+      return Pattern.compile(fileRegex);
```

Comment:
If `fileRegex` is null, a NPE will be throw. However, it should return `null` in such case instead.

Reply:
Putting in a new PR for this: https://github.com/cdapio/hydrator-plugins/pull/1005/files#r359378170